src/components/PupilComponents/ViewAllLessonsButton/ViewAllLessonsButton.tsx

```diff
@@ -4,12 +4,16 @@ import { resolveOakHref } from "@/common-lib/urls";
 
 type ViewAllLessonsButtonProps = {
   href?: string | null;
+  onClick?: () => void;
 };
 
 /**
  * A tertiary button to link back to the directory of lessons
  */
-export const ViewAllLessonsButton = ({ href }: ViewAllLessonsButtonProps) => {
+export const ViewAllLessonsButton = ({
+  href,
+  ...rest
```

Comment:
spreading the rest of the props is overkill when there's only one prop and it reduces readability. just destructure onClick as per normal.